core/src/main/java/com/linecorp/armeria/client/DefaultClientBuilderParams.java

```diff
@@ -70,4 +72,14 @@ public String toString() {
                           .add("type", type)
                           .add("options", options).toString();
     }
+
+    protected final URI cleanUri(final URI uri) {
+        if (!uri.getScheme().contains(SerializationFormat.NONE.uriText())) {
```

Comment:
How about this:

```java
final String nonePrefix = SerializationFormat.NONE.uriText() + '+';
if (!uri.getScheme().startsWith(nonePrefix)) { ... }
```

Reply:
I thought serialization protocol could be *none+http* or *http+none*, based on Scheme.SCHEMES map. Hence I had not checked for starts with, and replaced specific sequences.

Reply:
Ah, OK. Then how about reconstructing the URI at `DefaultClientFactory.newClient(URI, Class, ClientOptions)` if `uri.getScheme()` is not equal to the validated `scheme.uriText()`, so that the user-specified URI always has normalized scheme?

Reply:
That's a good idea. I have moved it to DefaultClientFactory. And I am using the derived Scheme to check for SerializationFormat.None (instead of uriText comparison).